src/core/server/status/status_service.test.ts

```diff
@@ -230,6 +237,51 @@ describe('StatusService', () => {
           ]
         `);
       });
+
+      it('allows more severe plugin statuses to take precendence over lower core statuses', async () => {
+        const pluginDependencies = new Map([
+          ['a', []],
+          ['b', ['a']],
+        ]);
+        const elasticsearch$ = new BehaviorSubject(available);
+        const savedObjects$ = new BehaviorSubject(available);
+        const setup = await service.setup({
+          elasticsearch: {
+            status$: elasticsearch$,
+          },
+          savedObjects: {
+            status$: savedObjects$,
+          },
+          pluginDependencies,
+        });
+
+        const pluginA$ = new BehaviorSubject(available);
+        setup.plugins.set('a', pluginA$);
+
+        const statusUpdates: ServiceStatus[] = [];
+        const subscription = setup.overall$.subscribe((status) => statusUpdates.push(status));
+
+        pluginA$.next(degraded);
+        subscription.unsubscribe();
+
+        expect(statusUpdates.map(({ level, summary }) => ({ level, summary })))
+          .toMatchInlineSnapshot(`
+          Array [
+            Object {
+              "level": available,
+              "summary": "All services are available",
+            },
+            Object {
+              "level": degraded,
+              "summary": "[a]: This is degraded!",
+            },
+            Object {
+              "level": degraded,
+              "summary": "[2] services are degraded",
+            },
```

Comment:
This highlights a current problem with this Observable-based architecture: a single update to one plugin's status may result in many overall updates as it propagates through the dependency tree.

I plan to experiment more tomorrow with RxJS's various schedulers and debouncing utilities, but I'm not sure yet there is a great solution to this. Even if there is, we still would need to provide guidance or utilities so that plugin authors can also avoid this same issue in their custom status implementations.

Reply:
https://www.learnrxjs.io/learn-rxjs/operators/filtering/debouncetime ?

Reply:
+1 for debounce effect of some kind to the `overall$` observable, while keeping the `derivedStatus$` exposed to plugins 'fresh'.

Reply:
My goal was to avoid any time-based approach in order to make the behavior very consistent for synchronous Observable sources. I was able to get this to work using `debounce` with a variable number of microtasks based on the number of known observables. Seems to be working correctly! Now the observables should only emit once per macro-task.

Reply:
I decided to just go with a time-based apporoach after all. Microtasks seemed like too much magic and very fragile to break. It's also easy to write a test which ends up changing the microtasks queue itself, leading to strange results in tests.

> We probably want to keep the derivedStatus$ exposed to plugins 'fresh' to avoid delay in status recomputation though?

+1 to that. I am only going to debounce the overall$ and plugins$ observables